lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/IO.hs

```diff
@@ -96,20 +104,37 @@ readWalletState WalletInstance{env,walletState} =
     Operations
     Initialization
 ------------------------------------------------------------------------------}
-withWallet :: WalletEnv IO -> (WalletInstance -> IO a) -> IO a
-withWallet env@WalletEnv{..} action = do
-    walletState <- loadWalletStateFromDatabase
+-- | Initialize a new wallet in the given environment.
+withWalletInit
+    :: XPub
+    -> Integer
+    -> WalletEnv IO
```

Comment:
can we move upwards `Wallet Env IO` here. I would argue this should be first argument as in the other functions. Just according to "least surprise" principle:-)

Reply:
Fair enough. 👍